homeassistant/components/control4/light.py

```diff
@@ -0,0 +1,219 @@
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyControl4.error_handling import C4Exception
+from pyControl4.light import C4Light
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_TRANSITION,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_TRANSITION,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from . import Control4Entity, get_items_of_category
+from .const import (
+    CONF_LIGHT_COLD_START_TRANSITION_TIME,
+    CONF_LIGHT_TRANSITION_TIME,
+    DOMAIN,
+)
+from .director_utils import director_update_data
+
+_LOGGER = logging.getLogger(__name__)
+
+CONTROL4_CATEGORY = "lights"
+CONTROL4_NON_DIMMER_VAR = "LIGHT_STATE"
+CONTROL4_DIMMER_VAR = "LIGHT_LEVEL"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up Control4 lights from a config entry."""
+    entry_data = hass.data[DOMAIN][entry.entry_id]
+    scan_interval = entry_data[CONF_SCAN_INTERVAL]
+    light_transition_time = entry_data[CONF_LIGHT_TRANSITION_TIME]
+    light_cold_start_transition_time = entry_data[CONF_LIGHT_COLD_START_TRANSITION_TIME]
+    _LOGGER.debug(
+        "Scan interval = %s, light transition time = %s, light cold start transition time = %s",
+        scan_interval,
+        light_transition_time,
+        light_cold_start_transition_time,
+    )
+
+    async def async_update_data_non_dimmer():
+        """Fetch data from Control4 director for non-dimmer lights."""
+        try:
+            return await director_update_data(hass, entry, CONTROL4_NON_DIMMER_VAR)
+        except C4Exception as err:
+            raise UpdateFailed(f"Error communicating with API: {err}")
+
+    async def async_update_data_dimmer():
+        """Fetch data from Control4 director for dimmer lights."""
+        try:
+            return await director_update_data(hass, entry, CONTROL4_DIMMER_VAR)
+        except C4Exception as err:
+            raise UpdateFailed(f"Error communicating with API: {err}")
+
+    non_dimmer_coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="light",
+        update_method=async_update_data_non_dimmer,
+        update_interval=timedelta(seconds=scan_interval),
+    )
+    dimmer_coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="light",
+        update_method=async_update_data_dimmer,
+        update_interval=timedelta(seconds=scan_interval),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await non_dimmer_coordinator.async_refresh()
+    await dimmer_coordinator.async_refresh()
+
+    items_of_category = await get_items_of_category(hass, entry, CONTROL4_CATEGORY)
+    for item in items_of_category:
+        if item["type"] == 7:
```

Comment:
What does 7 mean here?  Can we make it a constant so we show what the magic number represents?
```suggestion
        if item["type"] == MAYBE_SOME_CONSTANT:
```

Reply:
`7` is a Control4 item type that's analogous to Home Assistant entities (one device [type `6`] can have multiple entities [type `7`]). I'll probably add this to `const.py`